src/main/java/seedu/address/logic/commands/ClearCommand.java

```diff
@@ -2,22 +2,29 @@
 
 import static java.util.Objects.requireNonNull;
 
-import seedu.address.model.AddressBook;
+import java.util.List;
+
 import seedu.address.model.Model;
+import seedu.address.model.person.Person;
 
 /**
  * Clears the address book.
  */
 public class ClearCommand extends Command {
 
     public static final String COMMAND_WORD = "clear";
-    public static final String MESSAGE_SUCCESS = "Address book has been cleared!";
+    public static final String MESSAGE_SUCCESS = "Current list has been cleared!";
 
 
     @Override
     public CommandResult execute(Model model) {
         requireNonNull(model);
-        model.setAddressBook(new AddressBook());
+        List<Person> shownList = model.getFilteredPersonList();
+
+        while (shownList.size() != 0) {
+            Person personToDelete = shownList.get(0);
```

Comment:
The code works fine and is understandable, but would the '0' might be considered magic number? Perhaps a stream based approach could improve readability and changing it to `while(!shownList.isEmpty())` can improve clarity.